crates/cli/src/main.rs

```diff
@@ -275,6 +279,69 @@ mod linux {
     }
 }
 
+#[cfg(target_os = "linux")]
+mod flatpak {
+    use std::ffi::OsString;
+    use std::path::PathBuf;
+    use std::process::Command;
+    use std::{env, process};
+
+    /// Restarts outside of the sandbox if currently running within it
+    pub fn try_restart_to_host() {
+        if let Some(bin_dir) = get_flatpak_bin_dir() {
+            let mut args = vec!["/usr/bin/flatpak-spawn".into(), "--host".into()];
+            args.append(&mut get_xdg_env_args());
+            args.push("--env=ZED_IS_FLATPAK_INSTALL=1".into());
+            args.push(bin_dir.join("zed").into());
+
+            let mut is_app_location_set = false;
+            for arg in &env::args_os().collect::<Vec<_>>()[1..] {
+                args.push(arg.clone());
+                is_app_location_set |= arg == "--zed";
+            }
+
+            if !is_app_location_set {
+                args.push("--zed".into());
+                args.push(bin_dir.join("zed-app").into());
+            }
+
+            let error = exec::execvp("/usr/bin/flatpak-spawn", args);
+            eprintln!("failed restart cli on host: {:?}", error);
+            process::exit(1);
+        }
+    }
+
+    fn get_flatpak_bin_dir() -> Option<PathBuf> {
+        if let Ok(flatpak_id) = env::var("FLATPAK_ID") {
+            let install_dir = Command::new("/usr/bin/flatpak-spawn")
+                .arg("--host")
+                .arg("flatpak")
+                .arg("info")
+                .arg("--show-location")
+                .arg(flatpak_id)
+                .output()
+                .unwrap();
+            let install_dir = PathBuf::from(String::from_utf8(install_dir.stdout).unwrap().trim());
+            Some(install_dir.join("files").join("bin"))
+        } else {
+            None
+        }
+    }
+
+    fn get_xdg_env_args() -> Vec<OsString> {
+        let keep_keys = [
+            "XDG_DATA_HOME",
```

Comment:
While we want Zed’s config files to be in the flatpak xdg location, this is also breaking other software. Should find a nice way to keep the Zed files here but have the envs unset.

Reply:
I added some logic to `crates/util/src/paths.rs` to look for "FLATPAK_XDG_*_HOME". This means that all of Zed's stuff is in the flatpak location, but normal programs use the normal location.